From 7a55473bbe70d6f88563f021932966c8377b402d Mon Sep 17 00:00:00 2001
From: Thor Thayer <tthayer@opensource.altera.com>
Date: Wed, 27 Aug 2014 17:59:09 -0500
Subject: [PATCH 087/172] FogBugz #250531-1: fix l2 ECC enabling

Enable the l2 ECC in socfpga_init_irq. The l2 ECC bit must be turned on
before the l2 is enabled. Also, fix L2 cache build error when debug is
enabled.

Signed-off-by: Thor Thayer <tthayer@opensource.altera.com>
[czou:Original patch taken from
https://github.com/altera-opensource/linux-socfpga.git socfpga-4.1]
Signed-off-by: czou <cao.zou@windriver.com>
---
 arch/arm/mach-socfpga/socfpga.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-socfpga/socfpga.c b/arch/arm/mach-socfpga/socfpga.c
index 9ab8846..2e811f2 100644
--- a/arch/arm/mach-socfpga/socfpga.c
+++ b/arch/arm/mach-socfpga/socfpga.c
@@ -193,6 +193,7 @@ static void __init socfpga_init_irq(void)
 {
 	irqchip_init();
 	socfpga_sysmgr_init();
+	socfpga_init_l2_ecc();
 }
 
 static void socfpga_cyclone5_restart(enum reboot_mode mode, const char *cmd)
-- 
1.9.1

